src/pages/index.js

```diff
@@ -68,39 +91,135 @@ popupWithImage.setEventListeners();
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__subtitle",
+  avatarSelector: ".profile__avatar",
 });
 
+const deletePopup = new PopupWithForm("#modal-delete");
+deletePopup.setEventListeners();
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
+
+// handle image preview
+const handleImageClick = (cardData) => {
+  console.log(cardData);
+  popupWithImage.open(cardData);
+};
+
+// delete card
+const handleDeleteButton = (cardData) => {
+  deletePopup.open();
+  deletePopup.setSubmit(() => {
+    console.log(cardData);
+    api
+      .deleteCard(cardData._id)
+      .then(() => {
+        console.log("Card deleted successfully");
+        cardData.removeCard();
+        deletePopup.close();
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      });
+  });
+};
+
+//like a card
+
+const handleLikeButton = (cardData) => {
+  console.log(cardData.getLikeStatus());
+};
+
+function handleCardLike(card) {
+  api
+    .addLikes(card.getCardId())
+    .then(() => {
+      card.like();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleCardDislike(card) {
+  api
+    .removeLikes(card.getCardId())
+    .then(() => {
+      card.dislike();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
 // Functions
 
 function handleProfileFormSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
-  profilePopup.close();
+  profilePopup.renderLoading(true);
+  api
+    .updateProfileInfo(name, description)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      profilePopup.close();
+      profilePopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.renderLoading(false);
+    });
 }
 
 function handleAddCardFormSubmit({ name, link }) {
-  const cardElement = createCard({ name, link });
-  cardSection.addItem(cardElement);
-  cardPopup.close();
-  addCardForm.reset();
+  cardPopup.renderLoading(true);
+  api
+    .createNewCard({ name, link })
+    .then((data) => {
+      const cardElement = createCard(data);
+      cardSection.addItem(cardElement);
+      cardPopup.close();
+      formValidators["add-card-form"].resetValidation();
```

Comment:
`resetValidation` is not needed after submitting because you don't have errors in that case

Only `disableButton` is needed